### PR TITLE
World announcements

### PR DESCRIPTION
**Pull request by rixx**
- [x] data model
- [x] consumer
- [x] tests
- [x] documentation
- [ ] front-end
- [x] migrations (last to avoid conflicts with other model re-work)

### CODE REVIEW

**Comment by rixx** [comment]
Question: should we add visible announcements to the world state, or will the frontend retrieve ``announcement.list`` initially? @rashfael

**Comment by rashfael** [comment]
If the frontend will request the announcements on load anyways, I'd just put them in the world state instead.

**Comment by rashfael** [comment]
We discussed having announcements go through states to give them a defined lifecycle and avoid endless active->inactive loops, which will cause problems with client-side dismissal.
I was thinking of a state attribute with `draft` => `active` => `archived`. An announcement will only be visible when the state is `active` and `show_until` is valid. Feel free to choose other names for those states though.

To allow for easier recreation, we'll offer a cloning function, which can be done entirely in the frontend.
We also talked about having an moderator-only name/label for each announcement, but I have no real opinion on that yet.

**Comment by rixx** [comment]
reverted unintended force-push, please disregard

**Comment by rixx** [comment]
@rashfael you've got your states! New field is called "state", possible values "draft"/"active"/"archived", and state transitions are checked to be along allowed edges.

**Comment by rashfael** [comment]
Creating an announcement with `show_until` gives me
```
venueless-server-1  | 2022-02-27 21:44:13,332 ERROR 140608873027328 venueless.live.consumers consumers Encountered exception, close socket.
venueless-server-1  | Traceback (most recent call last):
venueless-server-1  |   File "/app/venueless/live/consumers.py", line 218, in dispatch
venueless-server-1  |     return await super().dispatch(message)
venueless-server-1  |   File "/usr/local/lib/python3.8/site-packages/channels/consumer.py", line 73, in dispatch
venueless-server-1  |     await handler(message)
venueless-server-1  |   File "/usr/local/lib/python3.8/site-packages/channels/generic/websocket.py", line 194, in websocket_receive
venueless-server-1  |     await self.receive(text_data=message["text"])
venueless-server-1  |   File "/usr/local/lib/python3.8/site-packages/channels/generic/websocket.py", line 257, in receive
venueless-server-1  |     await self.receive_json(await self.decode_json(text_data), **kwargs)
venueless-server-1  |   File "/app/venueless/live/consumers.py", line 171, in receive_json
venueless-server-1  |     await component.dispatch_command(content)
venueless-server-1  |   File "/app/venueless/live/modules/base.py", line 42, in dispatch_command
venueless-server-1  |     await self._commands[action](self, content[2])
venueless-server-1  |   File "/app/venueless/live/decorators.py", line 17, in wrapped
venueless-server-1  |     return await func(self, *args)
venueless-server-1  |   File "/app/venueless/live/decorators.py", line 136, in wrapped
venueless-server-1  |     return await func(self, *args)
venueless-server-1  |   File "/app/venueless/live/modules/announcement.py", line 24, in create_announcement
venueless-server-1  |     announcement = await create_announcement(
venueless-server-1  |   File "/usr/local/lib/python3.8/site-packages/asgiref/sync.py", line 444, in __call__
venueless-server-1  |     ret = await asyncio.wait_for(future, timeout=None)
venueless-server-1  |   File "/usr/local/lib/python3.8/asyncio/tasks.py", line 455, in wait_for
venueless-server-1  |     return await fut
venueless-server-1  |   File "/usr/local/lib/python3.8/concurrent/futures/thread.py", line 57, in run
venueless-server-1  |     result = self.fn(*self.args, **self.kwargs)
venueless-server-1  |   File "/usr/local/lib/python3.8/site-packages/channels/db.py", line 13, in thread_handler
venueless-server-1  |     return super().thread_handler(loop, *args, **kwargs)
venueless-server-1  |   File "/usr/local/lib/python3.8/site-packages/asgiref/sync.py", line 486, in thread_handler
venueless-server-1  |     return func(*args, **kwargs)
venueless-server-1  |   File "/app/venueless/core/services/announcement.py", line 10, in create_announcement
venueless-server-1  |     return Announcement.objects.create(
venueless-server-1  |   File "/app/venueless/core/models/announcement.py", line 42, in serialize_public
venueless-server-1  |     "show_until": self.show_until.isoformat() if self.show_until else None,
venueless-server-1  | AttributeError: 'str' object has no attribute 'isoformat'
```

**Comment by rixx** [comment]
I have no idea why that happened, and I have no patience to find out, so I fixed it with a giant, giant hammer – should be good now, @rashfael

File: docs/developer/api/announcements.rst [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+The announcements module allows organisers to push announcements to all users.
+Announcements can either have an expiry timestamp (``show_until``), or can
+be deactivated manually by the administrators.
+
+Receiving announcements
+-----------------------
+
+Announcements are always sent out with a ``created_or_updated`` message::
+
+    <= ["announcement.created_or_updated", {"id": "", "text": "", "show_until": "", "is_active": true}]
```
Comment (raphaelm) [inline docs/developer/api/announcements.rst:13]: is_active is no longer the current API this PR introduces

File: docs/developer/api/announcements.rst [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+To create an announcement, a user with the permission ``WORLD_ANNOUNCE`` sends
+a message like this::
+
+    => ["announcement.create", 1234, {"text": "Announcement text", "show_until": "timestamp or null", "is_active": true}]
+    <- ["success", 1234, {"announcement": []}]
+
+Setting ``is_active`` to ``false`` will keep the announcement hidden until it
+is published, otherwise it will be pushed to all users immediately.
+
```
Comment (raphaelm) [inline docs/developer/api/announcements.rst:37]: is_active is no longer the current API this PR introduces

**Pull request merged**